[RM Ch10] Update section 10.3.1 of RM (#1811)
cntt-n/CNTT: doc/ref_model/chapters/chapter10.md

This section addresses major open issues identified in the development of the Reference Model, Reference Architecture, Reference Implementation of the Common Cloud Infrastructure Lifecycle Framework. 

## 10.3.1 Discovery

> The workloads (VNFs/CNFs) and Cloud Infrastructure should be able to discover each other and exchange their capabilities required or offered. One of the key pain points for most of the operators is the VNF/CNF onboarding - both in terms of time and complexity. It could take weeks or months to onboard a VNF/CNF. There are lots of static and laborious checks performed to ensure the compatibility of the workloads with the corresponding Cloud Infrastructure. 

Reviewer: I would avoid that statement "It could take weeks or months to onboard a VNF/CNF. "
It doesn't refer to CPU time or execution time on the infrastructure, but manual preparation before the onboarding can start.
Also the extensive checks to ensure compatibility or workloads and infrastructure are not a disadvantage. But the need to use them by try and error is a problem. It should be solved by automation, which is said in the next sentence anyway.
Suggest:
```suggestion
The workloads (VNFs/CNFs) and Cloud Infrastructure should be able to discover each other and exchange their capabilities required or offered. One of the key pain points for most of the operators is the required preparation effort for VNF/CNF onboarding - both in terms of time and complexity.  
```

This section addresses major open issues identified in the development of the Reference Model, Reference Architecture, Reference Implementation of the Common Cloud Infrastructure Lifecycle Framework. 

## 10.3.1 Discovery

> The workloads (VNFs/CNFs) and Cloud Infrastructure should be able to discover each other and exchange their capabilities required or offered. One of the key pain points for most of the operators is the VNF/CNF onboarding - both in terms of time and complexity. It could take weeks or months to onboard a VNF/CNF. There are lots of static and laborious checks performed to ensure the compatibility of the workloads with the corresponding Cloud Infrastructure. 

Reviewer 2: Is the "weeks or months to onboard a VNF/CNF" due to an inability to discover "capabilities"? If not, then this doesn't belong here.

"checks performed to ensure the compatibility of the workloads with the corresponding Cloud Infrastructure" -- but in the case of "CNTT conformant workloads" many of these checks would already have been performed (onboarding success is the driver fro workload conformance).

for consistency:
```suggestion
The workloads (VNFs/CNFs) and Cloud Infrastructure should be able to discover each other and exchange their capabilities required or offered. One of the key pain points for most of the operators is workload onboarding - both in terms of time and complexity. It can take weeks or months to onboard a network function workload (VNF/CNF). There are a number of static and laborious checks that need to be performed to ensure the compatibility of the workloads with the Cloud Infrastructure. 
```

> The onboarding of the workloads (network functions) should be as much automated as possible. The CNTT conformant workloads and Cloud Infrastructure should be able to discover and negotiate their capabilities. Following should be supported: Cloud Infrastructure should be able to publish the capabilities it offers to workloads (network functions) and workloads should be able to query the Cloud Infrastructure for specific capabilities - such as number of cores, performance parameters. Negotiation/Hand Shake API - VNFs and NFVIs should be able to negotiate on certain capabilities. For instance, VNF desires HW acceleration for high throughput, but should be able to fall back to high throughput offered by NFVI via DPDK offering, and vice-a-versa.

Reviewer 2: ```suggestion
The onboarding of the workloads (network functions) should be automated as much as possible. The workloads and Cloud Infrastructure should be able to discover and negotiate their capabilities. Following should be supported: 
- Capabilities Discovery and Advertising
    - Cloud Infrastructure should be able to publish the capabilities it offers to workloads (network functions) 
    - workloads should be able to query the Cloud Infrastructure for specific capabilities - such as number of cores, performance parameters
- Capabilities Negotiation/Hand Shake API: 
    - workloads and Cloud Infrastructure should be able to negotiate on certain capabilities. For instance, workload desires HW acceleration for high throughput, but should be able to fall back to high throughput offered by Cloud Infrastructure via DPDK offering, and vice-a-versa.
```

> ## 10.3.2 Support Load Balance of VNF/CNFs [Initial language]

Reviewer 2: ```suggestion
## 10.3.2 Support Load Balance workloads (VNF/CNFs) [Initial language]
```

> Ability to load balance workflows through multiple instances of same VNF or CNF - e.g. using ECMP to distribute workloads through the multiple instances of Firewall. As an example imagine a distributed finance application with multiple instances of Web-tier and DB-tier. The traffic needs to flow through multiple instances of Firewall (for HA as well load balancing). There is no simple way to accomplish this in OpenStack.

Reviewer 2: What about the OSTK LBaaS - Octavia?

Author: @pgoyal01 Thanks for your suggestions for the content for 10.3.1 (Discovery). I have have committed those updates. 

This PR is to update section 10.3.1 only (please see the title of this PR). This does not update 10.3.2 or 10.3.4. Do you want to address those as part of this PR or a separate one? 
Please advise.

This section addresses major open issues identified in the development of the Reference Model, Reference Architecture, Reference Implementation of the Common Cloud Infrastructure Lifecycle Framework. 

## 10.3.1 Discovery [Initial language]

> The VNFs/CNFs and NFVI frameworks should be able to discover each other and exchange their capabilities required or offered. One of the key pain points for most of the operators is the VNF/CNF onbarding - both in terms of time and complexity. It could take weeks or months to onboard a VNF/CNF. There are lots of static and laborious checks performed to ensure the compatibility of the VNFs with the corresponding NFVI framework. 

Reviewer: ```suggestion
The VNFs/CNFs and NFVI frameworks should be able to discover each other and exchange their capabilities required or offered. One of the key pain points for most of the operators is the VNF/CNF onboarding - both in terms of time and complexity. It could take weeks or months to onboard a VNF/CNF. There are lots of static and laborious checks performed to ensure the compatibility of the VNFs with the corresponding NFVI framework. 
```

Reviewer: NFV uses the term "onboarding" for an automatic operation. Here we include the manual preparation time. Does that lead to confusion?

> The onboarding of the VNFs/CNFs should be as much automated as possible. The CNTT conformant VNFs/CBFs and NFVI frameworks should be able to discover and negotiate their capabilities. Following should be supported: NFVI should be able to publish the capabilities it offers to VNFs/CNFs and VNFs/CNFs should be able to query the NFVI for specific capabilities - such as number of cores, performance parameters. Negotiation/Hand Shake API - VNFs and NFVIs should be able to negotiate on certain capabilities. For instance, VNF desires HW acceleration for high throughput, but, should be able to fall back to high throughput offered by NFVI via DPDK offering, and vice-a-versa.

Reviewer 3: all VNFs/CNFs  should be replaced by workload 
The VNFs/CNFs and NFVI frameworks  ==> The workload and Cloud Infrastructure 
Typo The CNTT conformant VNFs/CBFs  ---- The CNTT conformant VNFs/CNFs 
compatibility of the VNFs ==> workload

Author: @ASawwaf I have uploaded the modified patch to make suggested changes

> The onboarding of the VNFs/CNFs should be as much automated as possible. The CNTT conformant VNFs/CBFs and NFVI frameworks should be able to discover and negotiate their capabilities. Following should be supported: NFVI should be able to publish the capabilities it offers to VNFs/CNFs and VNFs/CNFs should be able to query the NFVI for specific capabilities - such as number of cores, performance parameters. Negotiation/Hand Shake API - VNFs and NFVIs should be able to negotiate on certain capabilities. For instance, VNF desires HW acceleration for high throughput, but, should be able to fall back to high throughput offered by NFVI via DPDK offering, and vice-a-versa.

Reviewer: ```suggestion
The onboarding of the VNFs/CNFs should be as much automated as possible. The CNTT conformant VNFs/CNFs and NFVI frameworks should be able to discover and negotiate their capabilities. Following should be supported: NFVI should be able to publish the capabilities it offers to VNFs/CNFs and VNFs/CNFs should be able to query the NFVI for specific capabilities - such as number of cores, performance parameters. Negotiation/Hand Shake API - VNFs and NFVIs should be able to negotiate on certain capabilities. For instance, VNF desires HW acceleration for high throughput, but, should be able to fall back to high throughput offered by NFVI via DPDK offering, and vice-a-versa.
```

Do we need to explain that NFVI will expose its capabilities to VIM and VIM may select proper resources to match the requirements of VNF/CNF?

> ## 10.3.4 Packet Acceleration Request (e.g Hardware Acceleration) [Initial language]

Reviewer 2: what is this sentence trying to say, in particular the bolded part, "The acceleration hardware is typically optimized for specific kinds of workloads and **some form of disaggregation might be required to separate control and user plane responsibilities between _generic server hardware and accelerators_**."

Author: @pgoyal01 - Please see my comment above. I agree with you. Even I think this sentence is bit confusing. However, do you want me to address it as part of this PR or should there be a seperate PR for this? 
Note I did not write this section.

Reviewer 2: @sukhdevkapur Issue #1861 opened.
